packages/web3-eth-contract/src/contract.ts

```diff
@@ -961,31 +977,78 @@ export class Contract<Abi extends ContractAbi>
 	): ContractBoundMethod<T> {
 		return (...params: unknown[]) => {
 			let abiParams!: Array<unknown>;
-
-			try {
-				abiParams = validatorUtils.transformJsonDataToAbiFormat(abi.inputs ?? [], params);
-			} catch (error) {
-				throw new Web3ContractError(
-					`Invalid parameters for method ${abi.name}: ${(error as Error).message}`,
+			const abis = this._overloadedMethodAbis.get(abi.name)!;
+			let methodAbi: AbiFunctionFragment = abis[0];
+			if (abis.length === 1) {
+				try {
+					abiParams = validatorUtils.transformJsonDataToAbiFormat(
+						methodAbi.inputs ?? [],
+						params,
+					);
+				} catch (error) {
+					throw new Web3ContractError(
+						`Invalid parameters for method ${methodAbi.name}: ${
+							(error as Error).message
+						}`,
+					);
+				}
+				validator.validate(abi.inputs ?? [], abiParams);
+			} else {
+				const arrayOfAbis: AbiFunctionFragment[] = abis.filter(
+					_abi => (_abi.inputs ?? []).length === params.length,
 				);
-			}
 
-			validator.validate(abi.inputs ?? [], abiParams);
+				const errors: Web3ValidatorError[] = [];
+
+				for (const _abi of arrayOfAbis) {
+					try {
+						try {
+							abiParams = validatorUtils.transformJsonDataToAbiFormat(
+								_abi.inputs ?? [],
+								params,
+							);
+						} catch (error) {
+							throw new Web3ContractError(
+								`Invalid parameters for method ${_abi.name}: ${
+									(error as Error).message
+								}`,
+							);
+						}
+						validator.validate(
+							_abi.inputs as unknown as ValidationSchemaInput,
+							abiParams,
+						);
+						methodAbi = _abi;
+						break;
+					} catch (e) {
+						errors.push(e as Web3ValidatorError);
+					}
+				}
+				if (errors.length === arrayOfAbis.length) {
+					throw new Error(errors[0].message);
+				}
```

Comment:
Maybe we add a comment like "arguments don't match none of the overloaded messages" and show the possible inputs? I think, this will show the error for the first abi. What do you thinks?

Reply:
I've rewritten this part to send all errors not just first